File: remove_duplicate_char_string.py
# Remove Duplicate Character in String

class Solution:

	# O(n)
	def remove_duplicate(self, s):
		char_in_s = []
		seen = set()
		for c in s:
			if c not in seen:
				seen.add(c)
				char_in_s.append(c)

		return ''.join(char_in_s)

s = Solution()
string = "tree traversal"
print s.remove_duplicate(string)

File: remove_duplicate_char_string/remove_duplicate_char_string.php
<?php
// Remove Duplicate Character in String

class Solution {

    public function removeDuplicate($string) {
        $char_in_string = [];
        $seen = [];
        $chars = str_split($string);

        foreach($chars as $char) {
            if (!isset($seen[$char])) {
                $char_in_string[] = $char;
                $seen[$char] = 1;
            }
        }

        return $char_in_string;
    }

}

$string = "Tree Traversal";
$sol = new Solution();
print implode('',$sol->removeDuplicate($string));
